src/config_design.config

```diff
@@ -0,0 +1,95 @@
+%% Cloudant_benchmark config changes design review
+%% https://cloudant.fogbugz.com/f/cases/62351/Improve-basho_bench-config-system-to-meet-cloudant-requirements
+
+%% ------------------------------
+%% CONFIGURATION TEMPLATES: 
+%% Optional configuration template files that predefine config parameters (Name,Value,ValidatorFun,Description,etc) and preset default values in environment
+%% When provided, they would allow us to validate-on-load incoming parameters that match and warn about unknown config params that don't to catch typos
+%% Config template files could be nested, used to auto-generate a key-value only config file for local config file use, and would help dumping all config params to run log
+
+{config_templates, ["basho_bench", "cloudant_http", "http_driver"]}.
+
+%% GLOBAL parameters would work as they do now with validation if a template is provided
+%%    basho_bench_config:load uses Terms = file:consult(file) then stores these using application:set_env
+
+{option1, 1}.
+
+%% ------------------------------
+%% PER-OPERATIONS CONFIGURATION: https://github.com/cloudant/perf/issues/23 and 29
+%%
+%% Operations would change to allow optional per-operation keygen and valgen
+{operations, [
+   %% basho_bench_worker and _stats currently read this by looking for exactly two values - Name, Count
+   {get, 1},
+
+   %% Using alternate {op,[list]} format would keep 2-tuples so either simple or extended format could be used
+   %% QUESTION: Do we need ONLY keygen or the OPTION of BOTH keygen and valgen ? Would we ever need MORE per-op options? 
+
+   {put, [{count,1},  % Alternate way to show count in more complex data structure, existing lookups would need to be modified
+
+          % basho_bench_worker:init would turn these into new Keygen and ValueGen entities and
+	  %    then store those into worker State as per-op-config LIST(?) [{put, [{keygen, KeyGen}, {valgen, ValGen}, ...]}, ... ]
+          % basho_bench_worker:worker_next_op2 would check for per-op keygen/valgen, potentially push added config onto State
+
+          {key_generator, {int_to_str, {partitioned_sequential_int, 100000}}},
+          {value_generator, {function,http_couchdb_driver,make_json_doc,[]}}
+	 ]}
+   ]}.
+
+%% ------------------------------
+%% WORKER_GROUP SUPPORT: https://github.com/cloudant/perf/issues/32
+%%
+%% "worker_types" would provide per-worker config list, miniature copy of current config file for what is different from global
+{worker_types, [
+   %% Each worker_type would have its own list of config options to be used on top of global parameters known to basho_bench_config
+   {worker_name_a, [ {weight, 5}, 
+                     {driver, cb_driver}, 
+                     {operations, [
+                         {get,1},
+                         {put, [ {count,1}, 
+                                 {key_generator, keyGen1}
```

Comment:
It is a little confusing how I showed it but the intent had been simply to show how Operations with added config as one case and then as a DIFFERENT case how things would look if we were using workers. You would not usually specify Ops both ways in the same config file but would do one OR the other. 

The basic idea is that you could specify per-worker configuration information when needed it OTHERWISE it would use whatever it finds in the global configuration (top level definitions).  So if you specified a top level keygen/valgen that would be used UNLESS it was overridden per-operation, per-worker, or per-worker/per-operation.